.github/workflows/lint_and_test.yaml

```diff
@@ -0,0 +1,35 @@
+name: Python package
+
+on: [push]
```

Comment:
What is an expected outcome of adding this `.github/workflows/lint_and_test.yaml`? Will this be run by default on every push? I'm not sure for this file, so I'll double check with @sfc-gh-kbregula